localstack/services/events/provider_v2.py

```diff
@@ -35,15 +95,179 @@ def create_event_bus(
         tags: TagList = None,
         **kwargs,
     ) -> CreateEventBusResponse:
-        event_bus_arn = f"arn:aws:events:{context.region}:{context.account_id}:event-bus/{name}"
-        event_bus = {"Name": name, "Arn": event_bus_arn}
-        self._event_buses[name] = event_bus
+        region = context.region
+        account_id = context.account_id
+        store = self.get_store(context)
+        if name in store.event_buses.keys():
+            raise ResourceAlreadyExistsException(f"Event bus {name} already exists.")
+        event_bus_worker = self.create_event_bus_worker(
+            name, region, account_id, event_source_name, tags
+        )
+        store.event_buses[event_bus_worker.event_bus.name] = event_bus_worker.event_bus
 
         response = CreateEventBusResponse(
-            EventBusArn=event_bus_arn,
+            EventBusArn=event_bus_worker.arn,
         )
         return response
 
+    @handler("DeleteEventBus")
+    def delete_event_bus(self, context: RequestContext, name: EventBusName, **kwargs) -> None:
+        if name == "default":
+            raise ValidationException("Cannot delete event bus default.")
+        store = self.get_store(context)
+        try:
+            if event_bus := self.get_event_bus(name, store):
+                if event_bus_worker := self._event_bus_workers.pop(event_bus.arn):
+                    if rules := getattr(event_bus_worker, "rules", None):
```

Comment:
You can create and delete an event bus without adding any rules, therefore the rules will be not defined `rules: Optional[RuleDict] = None,`- therefore this save method to get it is required, else the test `test_create_list_describe_delete_custom_event_buses` would fail